src/builtin-addons/core/template-completion-provider.ts

```diff
@@ -365,39 +420,26 @@ export default class TemplateCompletionProvider {
     }
 
     if (this.hasNamespaceSupport) {
-      const registry = this.server.getRegistry(this.project.root);
-      const extraCompletions: CompletionItem[] = [];
-      const filteredCompletions = completions.filter((item) => {
-        if (item.detail === 'component') {
-          const paths = registry.component[normalizeToClassicComponent(item.label)] || [];
-          const roots = this.project.addonsMeta
-            .filter(({ root }) => {
-              return paths.find((p) => p.startsWith(root));
-            })
-            .sort((a, b) => {
-              return b.root.length - a.root.length;
-            });
+      const resultsMap = getResultsObj(this.project.addonsMeta, this.server, focusPath);
```

Comment:
before getting it, could we check is results has components at all (if not - we don't need to call function at all)
also, one more check - inside getResultsObj we could use already resolved component names (if registry for some addon return us new unmet component, we could ignore it)

Reply:
done